Neither i nor j are ever counted downward for the array pointers, so assume
they were intended to be signed integers.

Fixes a compiler warning on HPPA triggering an error because of
-Werror=strict-overflow -Wstrict-overflow=2:

src/connect.c:509:7: error: assuming signed overflow does not occur when
simplifying conditional to constant [-Werror=strict-overflow]
     if(j != 0)
	        ^
--- a/src/connect.c
+++ b/src/connect.c
@@ -471,7 +471,7 @@
     fd_set *readfds, struct timeval *timeout) {
   fd_set origfds;
   socket_t fd;
-  int i,j;
+  unsigned int i,j;
   int rc;
   int base_tm, tm;
   struct ssh_timestamp ts;
